### PR TITLE
working on bugs with css

### PR DESCRIPTION
**Pull request by Gangotri-sg3**

### CODE REVIEW

File: client/src/components/Address/CreateAddress.js [hunk 1]
```diff
@@ -110,28 +110,32 @@ const CreateAddress = () => {
...
-          phone={phone}
-          setPhone={setPhone}
-        />
+        <div style={{ marginLeft: "10%" }}>
+          <AddressForm
+            street={street}
+            setStreet={setStreet}
+            city={city}
+            setCity={setCity}
+            pin_code={pin_code}
```
Comment (iMagesh) [inline client/src/components/Address/CreateAddress.js:122]: you seem to have too many props sent to AddressForm, can we simplify that?
maybe put all the data related props in one object maybe?
like, street, pincode, state, city and etc could be passed as one object right?

File: server/routes/payment.js [hunk 2]
```diff
@@ -1,12 +1,7 @@
 const express = require("express");
 const router = express.Router();
 const knex = require("../utils/dbConfig");
-const Razorpay = require("razorpay");
-
-var razorpay = new Razorpay({
-  key_id: process.env.RAZOR_PAY_KEY,
-  key_secret: process.env.RAZOR_PAY_KEY_SECREAT,
-});
+const payment = require("../utils/paymentMode");
```
Comment (iMagesh) [inline server/routes/payment.js:4]: can we name it as paymentGateway instead of paymentMode?

File: client/package.json [hunk 3]
```diff
@@ -12,6 +12,7 @@
     "font-awesome": "^4.7.0",
     "razorpay-checkout": "^1.0.4",
     "react": "^17.0.2",
+    "react-avatar": "^4.0.0",
```
Comment (iMagesh) [inline client/package.json:15]: why are we using react-avatar for? can't we simply use avatar from bootstrap?
is this really necessary?

File: client/src/components/Address/CreateAddress.js [hunk 4]
```diff
@@ -32,7 +32,7 @@ const CreateAddress = () => {
   };
   let submitHandler;
   let id = location.pathname.slice(14);
-  if (location.pathname == "/user/address/null") {
+  if (location.pathname == "/user/address/new") {
```
Comment (iMagesh) [inline client/src/components/Address/CreateAddress.js:35]: this line seems to be duplicated multiple times within the same file, can we move it to a const or something like that and just call it in all the places?

**Comment by iMagesh** [comment]
@Gangotri-sg3 resolve the comments and let me know, lets merge this PR asap

File: server/routes/payment.js (hunk 2 above)
Comment (Gangotri-sg3) [reply server/routes/payment.js:4]: fiixed

File: client/package.json (hunk 3 above)
Comment (Gangotri-sg3) [reply client/package.json:15]: its not needed

Comment (Gangotri-sg3) [reply client/package.json:15]: I will remove it

File: client/src/components/Address/CreateAddress.js (hunk 4 above)
Comment (Gangotri-sg3) [reply client/src/components/Address/CreateAddress.js:35]: changed

File: client/src/components/Address/CreateAddress.js (hunk 1 above)
Comment (Gangotri-sg3) [reply client/src/components/Address/CreateAddress.js:122]: I didnot get it , can u show me some examples so i can try to implement?

**Comment by iMagesh** [comment]
@Gangotri-sg3 did you push the changes?

**Comment by Gangotri-sg3** [comment]
Fixed issues ,Please check once @iMagesh

File: client/src/components/Address/Address.js [hunk 5]
```diff
@@ -45,31 +40,39 @@ const Address = () => {
       })
       .catch((err) => {
         setMessage("error in razor pay");
-        nullMessage();
+        setTimeout(() => {
+          setMessage(null);
+        }, 6000);
```
Comment (iMagesh) [inline client/src/components/Address/Address.js:45]: These lines of code are repeated multiple times *duplication*
what can we do it avoid this duplication?

File: client/src/components/Address/Address.js [hunk 6]
```diff
@@ -143,7 +152,9 @@ const Address = () => {
       })
       .catch((err) => {
         setMessage("error in getting cart details");
-        nullMessage();
+        setTimeout(() => {
+          setMessage(null);
+        }, 6000);
```
Comment (iMagesh) [inline client/src/components/Address/Address.js:157]: This is bad, repeating the same lines of code in many places.

File: client/src/components/Address/Address.js [hunk 7]
```diff
@@ -164,12 +175,8 @@ const Address = () => {
           .then((res) => {
             postOrderItems(ordersID)
               .then(async (res) => {
-                if (res.data.message == "Sucessfully added order items!!") {
-                  await deleteCart(cartItems.cart_id);
-                  navigate("/order/confirm");
-                } else {
-                  navigate("/order/error");
-                }
+                await deleteCart(cartItems.cart_id);
```
Comment (iMagesh) [inline client/src/components/Address/Address.js:178]: what will happen if this async call fails?
are you handling the error case well?

File: client/src/components/Address/CreateAddress.js [hunk 8]
```diff
@@ -32,9 +34,59 @@ const CreateAddress = () => {
...
+        setErrorMessage("Please enter email");
+        setTimeout(() => {
+          setErrorMessage(null);
+        }, 6000);
+        return;
+      }
+      if (phone == null) {
+        setErrorMessage("Please enter phone");
+        setTimeout(() => {
+          setErrorMessage(null);
```
Comment (iMagesh) [inline client/src/components/Address/CreateAddress.js:57]: there are too many if conditions, can we do something to reduce that? or can we refactor this better? just asking

File: client/src/components/Address/CreateAddress.js (hunk 1 above)
Comment (iMagesh) [reply client/src/components/Address/CreateAddress.js:174]: there are too many props sent to the component right? can we reduce it? too many props don't look good, just asking if we can reduce it?

Comment (Gangotri-sg3) [reply client/src/components/Address/CreateAddress.js:174]: I tried to change,But I couldnot do ,I will check with mani or chaitanya,

File: client/src/components/Address/Address.js (hunk 7 above)
Comment (Gangotri-sg3) [reply client/src/components/Address/Address.js:178]: this I will remove and start from the scratch,as we change db design now.

File: client/src/components/Address/CreateAddress.js (hunk 8 above)
Comment (Gangotri-sg3) [reply client/src/components/Address/CreateAddress.js:57]: I thought of keeping all the conditions in one if using or operator,but messages are different so kept all separately.

File: client/src/utils/nullErrorMessage.js [hunk 9]
```diff
@@ -0,0 +1,6 @@
+export const nullMessage = (errorMessage) => {
```
Comment (iMagesh) [inline client/src/utils/nullErrorMessage.js:1]: will it be better if we change the name of this function to
`clearMessageTimeout` ?
does this name say more about what is going inside the function?

**Comment by iMagesh** [comment]
@Gangotri-sg3 can you quickly fix this? let's merge??

File: client/src/utils/nullErrorMessage.js (hunk 9 above)
Comment (Gangotri-sg3) [reply client/src/utils/nullErrorMessage.js]: Yes I changed

File: client/src/components/Address/CreateAddress.js (hunk 1 above)
Comment (iMagesh) [reply client/src/components/Address/CreateAddress.js:168]: We'll discuss that over a call

File: client/src/utils/imageData.js [hunk 10]
```diff
@@ -0,0 +1,12 @@
+const BASE_URL = process.env.REACT_APP_BASE_URL;
+
+let imageData = [
+  BASE_URL + "/data/image1.jpeg",
+  BASE_URL + "/data/image2.jpeg",
+  BASE_URL + "/data/image3.jpeg",
+  BASE_URL + "/data/image4.jpeg",
+  BASE_URL + "/data/image5.jpeg",
+  BASE_URL + "/data/image6.jpeg",
```
Comment (iMagesh) [inline client/src/utils/imageData.js:9]: All these images you have added are not HD. In my laptop it looks blury. Can we get 4k images ? need better quality @Gangotri-sg3

**Pull request merged**